x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -144,23 +188,80 @@ function DatatableComponent(props: DatatableProps & { formatFactory: FormatFacto
         columns={props.args.columns.columnIds
           .map(field => {
             const col = firstTable.columns.find(c => c.id === field);
+            const colIndex = firstTable.columns.findIndex(c => c.id === field);
+
+            const filterable = col?.meta?.type && props.getType(col.meta.type)?.type === 'buckets';
             return {
               field,
               name: (col && col.name) || '',
+              render: (value: unknown) => {
+                const formattedValue = formatters[field]?.convert(value);
+                if (filterable) {
+                  return (
+                    <EuiFlexGroup
+                      className="lnsDataTable__cell"
+                      data-test-subj="lnsDataTableCellValueFilterable"
+                      gutterSize="xs"
+                    >
+                      <EuiFlexItem grow={false}>{formattedValue}</EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        <EuiFlexGroup
+                          responsive={false}
+                          gutterSize="none"
+                          alignItems="center"
+                          className="lnsDataTable__filter"
+                        >
+                          <EuiToolTip
+                            position="bottom"
+                            content={i18n.translate('xpack.lens.filterForValueButtonTooltip', {
+                              defaultMessage: 'Filter for value',
+                            })}
+                          >
+                            <EuiButtonIcon
+                              iconType="plusInCircle"
+                              color="text"
+                              aria-label={i18n.translate(
+                                'xpack.lens.filterForValueButtonAriaLabel',
+                                {
+                                  defaultMessage: 'Filter for value',
```

Comment:
For the aria-labels only, I'm wondering if we should include the value here like `Filter for geo.src: AE`?

Reply:
done!